kotlinx-coroutines-core/common/src/flow/operators/Transform.kt

```diff
@@ -34,6 +35,11 @@ public inline fun <T> Flow<T>.filterNot(crossinline predicate: suspend (T) -> Bo
 @Suppress("UNCHECKED_CAST")
 public inline fun <reified R> Flow<*>.filterIsInstance(): Flow<R> = filter { it is R } as Flow<R>
 
+/**
+ * Returns a flow containing only values that are instances of the given [klass].
+ */
+public fun <R : Any> Flow<*>.filterIsInstance(klass: KClass<R>): Flow<R> = filter { klass.isInstance(it) } as Flow<R>
```

Comment:
Why not this signature?
```suggestion
public fun <R : T> Flow<T>.filterIsInstance(klass: KClass<R>): Flow<R> = filter { klass.isInstance(it) } as Flow<R>
```

Reply:
Ah, nevermind, got it: we may just have a `KClass<*>`.

Reply:
Not sure what you mean, your suggestion as is leads to unresolved references, I might be missing something

I used `<R : Any>` because `KClass` requires an upper bound: `interface KClass<T : Any>`

Reply:
I mistyped initially, you're probably replying to the old version. I was asking why not add the requirement that the passed `KClass` inherits the elements in the `Flow`.

Reply:
(yeah, sorry, for some reason github didn't load any additional comments from you :\ )